strutil/pathiter.go

```diff
@@ -52,33 +52,40 @@ type PathIterator struct {
 // The path is passed through filepath.Clean automatically.
 func NewPathIterator(path string) (*PathIterator, error) {
 	cleanPath := filepath.Clean(path)
-	if cleanPath != path && cleanPath+"/" != path {
+	if cleanPath != path {
```

Comment:
I'm not entirely sure this change is safe, as it's hard to tell if all the code paths reaching here do really clean end "/", I see an entry like this

/var/lib/snapd/mount/snap.chromium.fstab:/var/snap/cups/common/run /var/cups/ none bind,rw 0 0

I'm not sure if that /var/cups/ there will hit here or not?

osutil.ParseMountEntry doesn't seem to do Clean in itself

It's slightly clearer for the cases in interfaces/apparmor but not clear for the cases in snap-update-ns.

Maybe @zyga can comment on what was the original intention here

Reply:
I don't recall exactly, but I would guess that I didn't want to require a particular go semantics of a "clean" path given that apparmor and snap-update-ns both have a specific meaning and the trailing slash is semantically important.

I think I would prefer to keep this as-is as otherwise we just regress on this feature of the path iterator IMO.

Reply:
Sounds fabulous. So assuming this sounds good, I'll just have it accept paths with an ending slash and then just never return it unless the user calls the Path method. CurrentPath, CurrentBase, CurrentDir will all never return that ending slash.